drivers/gpio/gpio_mcux_igpio.c

```diff
@@ -121,6 +121,18 @@ static int mcux_igpio_pin_interrupt_configure(struct device *dev,
 	u8_t icr;
 	int shift;
 
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		key = irq_lock();
+
+		WRITE_BIT(base->ISR, pin, 1);
```

Comment:
I don't have the reference manual but I assume we write to the ISR register to clear pending interrupts? This is not required since we're about to disable the interrupt. We need to clear pending interrupts only before we enable them.